salt/_grains/caasp_nodename.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/python
+
+import salt.grains.core
+
+def hostname():
+    """
+    Override core.hostname grain to provide lowercased items
+    localhost, fqdn, host, domain
+    """
+    grains = salt.grains.core.hostname()
+    grains['domain'] = grains['domain'].lower()
+    grains['fqdn'] = grains['fqdn'].lower()
+    grains['host'] = grains['host'].lower()
+    grains['localhost'] = grains['localhost'].lower()
+    return grains
```

Comment:
Instead of returning a dictionary with many items that are already in the global grains, I think this could be optimized and return a dictionary with only the overridden grains. Something like:

```Python
    grains = salt.grains.core.hostname()
    return {
        'domain': grains['domain'].lower(),
        'fqdn': grains['fqdn'].lower(),
        'host': grains['host'].lower(),
        'localhost': grains['localhost'].lower()
    }
```

Reply:
We currently use `grains['nodename']` everywhere - so that's the one we most need to override!

Reply:
Right, `nodename` is the most important grain for us...

Reply:
Aside from this, we need to double check, because without checking it looks to me that when the salt-minion starts (without having synced grains with the master), it will run its mine functions (https://github.com/kubic-project/salt/blob/143b3e663a4a8bc1f3fe3e2835156e82238168b2/pillar/mine.sls#L5-L7), and probably publish on the mine the `nodename` containing upper letters (we use a lot the pattern `mine.get` with `fun=nodename`).